stream-feeds-android-client/src/main/kotlin/io/getstream/feeds/android/client/internal/client/reconnect/FeedWatchHandler.kt

```
@@ -0,0 +1,71 @@
+ *
+ *    https://github.com/GetStream/stream-feeds-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getstream.feeds.android.client.internal.client.reconnect
+
+import io.getstream.android.core.api.model.connection.StreamConnectionState
+import io.getstream.feeds.android.client.api.model.FeedId
+import io.getstream.feeds.android.client.api.state.query.FeedQuery
+import io.getstream.feeds.android.client.internal.repository.FeedsRepository
+import java.util.concurrent.ConcurrentHashMap
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.joinAll
+import kotlinx.coroutines.launch
+
+/**
+ * Handles re-watching feeds upon reconnection.
+ *
+ * Keeps track of feeds that are being watched and re-subscribes to them when the connection is
+ * re-established.
+ *
+ * @property connectionState A [Flow] that emits events when the connection state changes to
+ *   [StreamConnectionState.Connected].
+ * @property feedsRepository The [FeedsRepository] used to rewatch feeds on connection.
+ * @property scope The [CoroutineScope] in which to launch coroutines for re-watching feeds.
+ */
+internal class FeedWatchHandler(
+    private val connectionState: Flow<StreamConnectionState>,
+    private val feedsRepository: FeedsRepository,
+    scope: CoroutineScope,
+) {
+    private val watched = ConcurrentHashMap<FeedId, Unit>()
+
+    init {
+        scope.launch {
+            connectionState.filterIsInstance<StreamConnectionState.Connected>().collect {
+                rewatchAll()
```

`rewatchAll()` swallows the potential failure of the request.

---

I know but what's the alternative? I'm going to change it to `queryFeeds` as you suggested above, so we can easily return a `Result`, but we're still executing this code in background automatically, so clients won't receive anything anyway. Maybe we can log failures?